docs/export-queue.md

```diff
@@ -0,0 +1,259 @@
+    //represents the external query system we want to update from Fluo
+    QuerySystem querySystem;
+
+    @Override
+    protected void processExports(Iterator<SequencedExport<Bytes, CountUpdate>> exportIterator) {
+      BatchUpdater batchUpdater = querySystem.getBatchUpdater();
+      while(exportIterator.hasNext()){
+        SequencedExport<Bytes, CountUpdate> exportEntry =  exportItertor.next();
+        Bytes row =  exportEntry.getKey();
+        UpdateCount uc = exportEntry.getValue();
+        long seqNum = exportEntry.getSequence();
+
+        String oldCountRow = String.format("%06d", uc.oldCount);
+        String newCountRow = String.format("%06d", uc.newCount);
+
+        //add a new entry to the inverted index
+        batchUpdater.insertRow(newCountRow, row, seqNum);
+        //remove the old entry from the inverted index
+        batchUpdater.deleteRow(oldCountRow, row, seqNum);
+      }
+
+      //flush all of the updates to the external query system
+      batchUpdater.close();
+    }
+  }
+```
+
+## Concurrency
+
+Additions to the export queue will never collide.  If two transactions add the
+same key at around the same time and successfully commit, then two entries with
+different sequence numbers will always be added to the queue.  The sequence
+number is based on the start timestamp of the transactions.
+
+If the key used to add items to the export queue is deterministically derived
+from something the transaction is writing to, then that will cause a collision.
+For example consider the following interleaving of two transactions adding to
+the same export queue in a manner that will collide. Note, TH1 is shorthand for
+thread 1.
+
+ 1. TH1 : val1 = g(transacion1.get(`row1`,`fam1:qual1`), transacion1.get(`rowA`,`fam1:qual2`))
+ 1. TH1 : key1 = f(`row1`,`fam1:qual1`)
+ 1. TH1 : exportQueueA.add(transaction1, key1, val1)
+ 1. TH2 : val2 = g(transaction2.get(`row1`,`fam1:qual1`), transacion1.get(`rowB`,`fam1:qual2`))
+ 1. TH2 : key2 = f(`row1`,`fam1:qual1`)
+ 1. TH2 : exportQueueA.add(transaction2, key2, val2)
+ 1. TH1 : transaction1.set(`row1`,`fam1:qual1`, val1)
+ 1. TH2 : transaction2.set(`row1`,`fam1:qual1`, val2)
+
+In the example above only one transaction will succeed because both are setting
+`row1 fam1:qual1`.  Since adding to the export queue is part of the
+transaction, only the transaction that succeeds will add something to the
+queue.  If the funtion f() in the example is deterministic, then both
+transactions would have been trying to add the same key to the export queue.
+
+The example below is a slight modification of the example above.  In this
+example both transactions will successfully add entries to the queue using the
+same key.  Both transactions succeed because they are writing to different
+cells (`rowB fam1:qual2` and `rowA fam1:qual2`). This may be ok depending on
+the application.  Keep this case in mind when using the export queue.  If
+uncertain, then follow the model in the example above.
```

Comment:
I suppose I should expand on that.  I am thinking of [Serializability][1].  When following the first model you know that if the same key exists multiple times in the queue with different sequence numbers that the transactions adding it to the queue executed serially (no overlap in time).   Reasoning about serial transaction is much easier than ones that could possibly overlap in time. 

I'll take a crack at adding something.

[1]: https://en.wikipedia.org/wiki/Serializability